### ISSUE TITLE
Cannot login using Safari’s auto-fill

### ISSUE THREAD

**Issue by timvonwerne** [home-assistant/frontend#16976]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When trying to login to my home assistant instance, the login fails when the username and password were automatically filled in by Safari although the data is correct. It works only if I manually enter the login credentials OR if I go into the fields, remove the last character and enter it again manually.

### Describe the behavior you expected

If the correct credentials are entered, either manually or automatically, I should be redirected to the dashboard.

### Steps to reproduce the issue

1. Go to the login screen of a home assistant instance
2. Let Safari automatically fill in the form

### What version of Home Assistant Core has the issue?

2023.6.1

### In which browser are you experiencing the issue with?

Safari

### Which operating system are you using to run this browser?

iOS 17 but also happens on iOS 16

**Comment by wallyto** [comment]
I am having the same issue. Only in Safari, on Mac and IOS devices. Chrome works fine with autofill.
With safari keychain or bit warden, autofill fails. I can copy and paste the password from bit warden and it works.
This has been going on for at least since March 2023... maybe longer.
Home Assistant 2023.6.3
Supervisor 2023.06.2
Operating System 10.3
Frontend 20230608.0 - latest

MacOS Ventura Version 13.3.1(a)
IOS Version 16.5.1
Bitwarden Version 2023.5.1 (9424)

**Comment by timvonwerne** [comment]
Just to add some more details, the problem seems to be in the transmission of the input values to the API. If the fields were automatically filled in by Safari, the following request is sent to `/auth/login_flow/{flow_id}`:

```json
{
    "username": "",
    "password": "",
    "client_id": "{client_id}"
}
```

Once the values have been changed manually, they appear to be saved correctly and transmitted in the request. I can also confirm that it is a Safari-only issue. With Chrome, the login works without problems.

**Comment by steverep** [comment]
If the autofill is failing, see #10274.

If the autofill succeeds yet the login fails, that's a separate issue. @timvonwerne please confirm that's what you are experiencing and which password manager you are using.

Seems like perhaps a bug in the polyfill function.  I'll check it out.

**Comment by wallyto** [comment]
This happens with either mac default auto fill from keychain, or auto fill from bitwarden. (All are the latest release). if I do copy/paste of user and pass from either, it will work properly. Therefore the passwords are correct. It is the auto fill. that doesn't work even though it shows the user/pass being filled in, before sending to HAS.

**Comment by steverep** [comment]
I don't have a Mac to test.  If you can, please investigate the `value` properties for each input field after the autofill.  There will be 2 sets of username/password fields: one  deep in shadow DOM and one in the light DOM.

**Comment by lu-mueller** [comment]
I have the same problem. In Safari on macOS when using Apple Keychain. Autofilling TOTP doesn't work either. I can copy and paste the username, password and TOTP and it works. This hasn't worked since I installed Home Assistant OS a week ago.

Home Assistant 2023.6.3
macOS 13.4.1

**Comment by timvonwerne** [comment]
@steverep Safari doesn't seem to offer an option to copy the shadow DOM's content, unfortunately. I hope the attached screenshots include the info you need to further investigate the issue.

![home-assistant_username-field](https://github.com/home-assistant/frontend/assets/27087655/1ce2c6f2-0204-4a9d-bbd1-bf6809ff1840)
![home-assistant_pw-field](https://github.com/home-assistant/frontend/assets/27087655/1904ca14-3093-47bd-b435-667a3d8482c5)

**Comment by kode54** [comment]
Tested on Safari on iPadOS 16. The forms work if I manually type in the password. The forms do not allow me to even paste into them with the onscreen keyboard or with the long press Paste button that appears onscreen.

**Comment by steverep** [comment]
@Roos-AID the problem you described in https://github.com/home-assistant/frontend/issues/10274#issuecomment-1670190085 is this issue.  This issue is that autofill works, but the username and password are not being sent for authentication.  
By contrast, #10274 is tracking problems where certain password managers do not autofill at all.

**Comment by Roos-AID** [comment]
@steverep Steve, thanks. You are right. 
Is there a developer in Netherlands that needs a Macbook with Safari? I might be able to provide one temporarily to support the debugging or I can help if some instructions are given to provide detailed logging or Safari debug info.

**Comment by Alexivia** [comment]
I do not have much experience with debugging browsers, but I am in NL and I have a macbook. I would be glad to help 🙂

**Comment by alanpearce** [comment]
It works if I add/remove a space to both username and password fields after auto-filling.

**Comment by bramkragten** [comment]
The change events are not fired it seems

**Comment by blue68camaro-git** [comment]
Manually adding in user name and and password and NOT using auto fill works, using Safari on macOS an iOS. Firefox works great with auto fill from keychain.

**Comment by dgabrysch** [comment]
Same here with iOS.

**Comment by DavidPhillipOster** [comment]
I'm experiencing this issue. Thanks for the workaround (erase, then manually re-type the last character of the password that were auto-filled in from Safari’s password autofill popup menu).

I'm on:
Home Assistant 2023.8.4
Supervisor 2023.08.1
Operating System 10.5
Frontend 20230802.1 - latest
Safari Version 16.6 (18615.3.12.11.2)
macOS Ventura 13.5.1 (22G90)

and the problem does **not** occur with ssh, Chrome, the mac HomeAssistant appStore app, or the iOS HomeAssistant app. Only Safari has the problem.

**Comment by steverep** [comment]
This should be fixed in 2023.9
